server/src/main/java/org/apache/druid/server/coordinator/duty/CompactSegments.java

```diff
@@ -296,6 +314,16 @@ static int findMaxNumTaskSlotsUsedByOneCompactionTask(@Nullable ClientCompaction
     }
   }
 
+  /**
+   * Returns the maximum number of task slots used by one MSQ compaction task at any time when the task is
+   * issued with the given context.
+   */
+  static int findMaxNumTaskSlotsUsedByOneMsqCompactionTask(@Nullable Map<String, Object> context)
+  {
+    return QueryContext.of(context).getInt(ClientMsqContext.CTX_MAX_NUM_TASKS, ClientMsqContext.DEFAULT_MAX_NUM_TASKS);
```

Comment:
Building a `QueryContext` object seems overkill and also a little confusing here.
Maybe we should just use `context.getOrDefault()` and the cast the result as an integer.

Reply:
nit drive by comment: `ClientMsqContext` should be `ClientMSQContext`

Reply:
Updated.